src/lib/serviceRoutines/postQueryContext.cpp

```diff
@@ -406,71 +399,81 @@ std::string postQueryContext
       requestV.push_back(requestP);
     }
 
-    for (unsigned int aIx = 0 ; aIx < cerP->contextElement.contextAttributeVector.size(); ++aIx)
+    //
+    // What if the Attribute Vector of the ContextElementResponse is empty?
+    // For now, just push it into localQcrsP, but only if its local, i.e. its contextElement.providingApplicationList is empty
+    //
+    if ((cerP->contextElement.contextAttributeVector.size() == 0) && (cerP->contextElement.providingApplicationList.size() == 0))
```

Comment:
The diff is really bad ...
The change is basically:
* queryForward now returns bool and does not touch the errorCode
* Before starting the loop over `cerP->contextElement.contextAttributeVector.size()`, I check for local contextElements with empty contextAttributeVector

The change in indentation (for for-block is now under 'else', so more indentation is inevitable) made the diff too hard, unfoirtunately :(